File: src/main/java/com/qsoftware/utilities/datetime/TimeSpan.java
package com.qsoftware.utilities.datetime;

import com.qsoftware.utilities.core.exceptions.InvalidOrderException;

import java.io.Serializable;

public class TimeSpan implements Serializable {
    private final DateTime from;
    private final DateTime to;

    public TimeSpan(DateTime from, DateTime to) {
        if (from.compareTo(to) > 0) throw new InvalidOrderException("Parameter ‘from’ is later than ‘to’.");

        this.from = from;
        this.to = to;
    }

    public boolean contains(DateTime dateTime) {
        return dateTime.isBetween(from, to);
    }

    public Duration toDuration() {
        return new Duration(to.toEpochSeconds() - from.toEpochSeconds());
    }
}

File: src/main/kotlin/qtech/utilities/datetime/TimeSpan.kt
package qtech.utilities.datetime

import qtech.utilities.core.exceptions.InvalidOrderException
import java.io.Serializable

class TimeSpan(from: DateTime, to: DateTime) : Serializable {
    private val from: DateTime
    private val to: DateTime
    operator fun contains(dateTime: DateTime): Boolean {
        return dateTime.isBetween(from, to)
    }

    fun toDuration(): Duration {
        return Duration((to.toEpochSeconds() - from.toEpochSeconds()).toDouble())
    }

    init {
        if (from > to) throw InvalidOrderException("Parameter ‘from’ is later than ‘to’.")
        this.from = from
        this.to = to
    }
}